book/bayesian_optimization.qmd

```diff
@@ -0,0 +1,844 @@
+Evolutionary strategies for example maintain a so-called population and generate new points for evaluation by choosing parents from that population and performing recombination operators such as mutation and crossover to generate the offspring of the next generation.
+In general, evolutionary strategies are very suited for black box optimization, however, if the overhead of the evaluation of the black box function becomes large (e.g., as in HPO), sample efficiency of an optimizer becomes highly relevant.
+Bayesian optimization (BO) describes a class of sample efficient iterative global black box optimization algorithms that make use of a so-called surrogate trained on observed data modelling the black box function.
+Based on this (non-linear) regression model and a so-called acquisition function quantifying the attractiveness of each point in the search space, the next candidate point is obtained that is to be evaluated next.
+After evaluating this candidate, the surrogate model is updated or re-trained and the process repeats itself until a termination criteria is met.
+Often, using BO results in very good optimization performance, especially if the overhead of the black box evaluation becomes large and optimization budget is tight.
+
+This chapter is of general relevance to users concerned with sample efficient black box optimization and HPO.
+In the following, we will give a brief general introduction to black box optimization making use of the `r bbotk`\index{bbotk} package.
+We then show how BO can be performed within the mlr3 ecosystem making use of the `r mlr3mbo`\index{mlr3mbo} package.
+Detailed introductions to black box optimization and BO are given in @hpo_practical, @hpo_automl and @garnett_2022.
+
+## Black Box Optimization {#sec-black-box-optimization}
+
+The `r bbotk` package is the workhorse package for black box optimization within the mlr3 ecosystem.
+At the heart of the package are the R6 classes:
+
+* `r ref("OptimInstanceSingleCrit")` and `r ref("OptimInstanceMultiCrit")`, which are used to construct an optimization 'instance' which describes the optimization problem and stores the results; and
+* `r ref("Optimizer")` which is used to get and set optimization algorithms.
+
+This should sound very familiar.
+Indeed, in @sec-optimization, the classes `r ref("TuningInstanceSingleCrit")`, `r ref("TuningInstanceMultiCrit")` and `r ref("Tuner")` were already introduced.
+Those three R6 classes essentially build upon `r ref("OptimInstanceSingleCrit")`, `r ref("OptimInstanceMultiCrit")` and `r ref("Optimizer")` which are the more general black box optimization base classes.
+
+Our running example will be to optimize the following sinusoidal function which is characterized by two local minima and one global minimum:
+$f: [0, 1] \rightarrow \mathbb{R}, x \mapsto 2x + \sin(14x)$.
+The local optima can pose a risk for many optimization algorithms to get stuck.
```

Comment:
Reference fig below, point to local minima and briefly explain why this means algs might get stuck